module/friendship/adapter/postgres/repository/friendship_test.go

```diff
@@ -0,0 +1,205 @@
+package repository
+
+import (
+	"context"
+	"testing"
+
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/adapter/postgres/model"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
+	"github.com/phantranhieunhan/s3-assignment/pkg/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type FriendshipTestCase struct {
+	name                    string
+	modifyFriendship        domain.Friendship
+	IsErrExistingFriendship bool
+	hasError                bool
+}
+
+func TestFriendship_Create(t *testing.T) {
+	ctx := context.Background()
+	suite := NewSuite(ctx)
+	repo := NewFriendshipRepository(suite.db)
+
+	cs := []FriendshipTestCase{
+		{
+			name: "successful",
+		},
+		{
+			name: "fail by reference user id",
+			modifyFriendship: domain.Friendship{
+				UserID: "user-id",
+			},
+			hasError: true,
+		},
+		{
+			name: "fail by reference subscriber id",
+			modifyFriendship: domain.Friendship{
+				FriendID: "friend-id",
+			},
+			hasError: true,
+		},
+	}
+	for _, tc := range cs {
+		t.Run(tc.name, func(t *testing.T) {
+			fri := domain.Friendship{
+				UserID:   util.GenUUID(),
+				FriendID: util.GenUUID(),
+				Status:   domain.FriendshipStatusFriended,
+			}
+			suite.prepareFriendship(t, ctx, fri)
+			created := fri
+			if tc.hasError {
+				if tc.modifyFriendship.FriendID != "" {
+					created.FriendID = "friend-id"
+				} else if tc.modifyFriendship.UserID != "" {
+					created.UserID = "user-id"
+				}
+			}
+			var err error
+
+			fri.Id, err = repo.Create(ctx, created)
+			if tc.hasError {
+				assert.Error(t, err)
+				assert.Empty(t, fri.Id)
+			} else {
+				assert.NoError(t, err)
+				assert.NotEmpty(t, fri.Id)
+			}
+
+			suite.rollbackFriendship(t, ctx, fri)
+		})
+
+	}
+}
+
+func TestFriendship_UpdateStatus(t *testing.T) {
+	ctx := context.Background()
+	suite := NewSuite(ctx)
+	repo := NewFriendshipRepository(suite.db)
+
+	cs := []FriendshipTestCase{
+		{
+			name: "successful",
+		},
+		{
+			name: "fail by invalid id",
+			modifyFriendship: domain.Friendship{
+				Base: domain.Base{Id: "sub-id"},
+			},
+			hasError: true,
+		},
+	}
+	for _, tc := range cs {
+		t.Run(tc.name, func(t *testing.T) {
+			fri := domain.Friendship{
+				UserID:   util.GenUUID(),
+				FriendID: util.GenUUID(),
+				Status:   domain.FriendshipStatusFriended,
+			}
+			suite.prepareFriendship(t, ctx, fri)
+			var err error
+			fri.Id, err = repo.Create(ctx, fri)
+			assert.NoError(t, err)
+			id := fri.Id
+			if tc.modifyFriendship.Id != "" {
+				id = tc.modifyFriendship.Id
+			}
+
+			updatedErr := repo.UpdateStatus(ctx, id, domain.FriendshipStatusBlocked)
+			result, err := repo.GetFriendshipByUserIDs(ctx, fri.FriendID, fri.UserID)
```

Comment:
No need to double check like this, data integrity can be checked via integration test. Unit test should focus on data flow

Reply:
Here, we have had integration test yet, so I believe that I can keep it here for this situation